src/utils/word-api.ts

```diff
@@ -17,42 +30,31 @@ class WordApi {
     });
   }
 
-  static async getJabRefCitations(): Promise<Array<Word.ContentControl> | void> {
-    return Word.run(async (context: Word.RequestContext) => {
-      const { contentControls } = context.document;
-      context.load(contentControls, "tag, length");
-      await context.sync();
-      return contentControls.items.filter(
-        (citation) =>
-          citation.tag.includes("JABREF-CITATION") &&
-          !citation.tag.includes("NEW")
-      );
-    }).catch((error) => {
-      console.log(`Error: ${JSON.stringify(error)}`);
-      if (error instanceof OfficeExtension.Error) {
-        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
-      }
-    });
+  static async getJabRefCitations(
+    context: Word.RequestContext
+  ): Promise<Array<Word.ContentControl>> {
+    const { contentControls } = context.document;
+    context.load(contentControls, "tag, length");
+    await context.sync();
+    return contentControls.items.filter((citation) =>
+      citation.tag.includes("JABREF-CITATION")
+    );
   }
 
   static async getPositionOfNewCitation(): Promise<number | void> {
     return Word.run(async (context: Word.RequestContext) => {
-      const { contentControls } = context.document;
-      context.load(contentControls, "tag, length");
+      const currentPosition = context.document.getSelection();
```

Comment:
Return type of the method should be without void if I'm not mistaken

Reply:
ts complains about return type without void
`Type 'number | void' is not assignable to type 'number'.
  Type 'void' is not assignable to type 'number'.ts(2322)`

Reply:
What does it mean when the method throws an error (since we are catching and logging it)?

The `catch` is what returns `void` since the error isn’t rethrown and no other value is returned.